client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -142,7 +142,9 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
 
           const constrained = constrain(parsed, decimalLimit, min, max);
 
-          setTextValue(formatValue(constrained));
+          // DON'T use the formatValue callback here, we only want to do FULL
+          // formatting onBlur
+          setTextValue(format(constrained));
```

Comment:
Someone had changed this with the `formatValue` callback was created. Have added comment here to prevent this happening again.

Reply:
There's a good reason for the change!
This is [the PR](https://github.com/msupply-foundation/open-msupply/pull/3322/files#diff-db59deaccdff071a56ae20033716d4aa22e8618c6b082e9a65adf640d55541c6)

The issue is that the NumericTextInput is forcing a numeric formatting - and this is not desired when formatting a port number (e.g. `9100` which needs to be constrained to a number, but not formatted to be `9,100` which gives a different meaning)

Reply:
I didn't remove the `noFormatting` option -- I just prevented it from being updated every keystroke and keeping in `onBlur`. Because otherwise, when you have a fixed decimal value (`decimalMin` prop), as soon as you enter a number, it get re-formatted to the "full" version (i.e `5` becomes `5.00`), which makes adding additional characters very difficult (and fails the tests).

Reply:
Okay, I think this solves both problems -- i.e. allows unformatted entry, but fixes the problem introduced by your original change: a2d122a33

Reply:
yes, but by adding the call to `format` back in, you get this:

<img width="734" alt="image" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/0aecae68-209b-4479-ae08-18d192622702">

because `format` applies the formatting as I type. True, it is cleared on blur, but it is disconcerting to use the input.

Can you instead please change the line to this:

```
setTextValue(noFormatting ? String(constrained) : format(constrained));
```

the issue is that format is called with no parameters on keypress, but is called with params on blur, so there's differing behaviour.

Reply:
Okay, I think you missed the last comment which fixed both, but I've changed it again to your suggestion as it's a bit more thorough (and closer to original behaviour): 85ca54352